### PR TITLE
[web][api][SIMS #400]Student File Upload and saved to database

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Allow the student to upload files to the server and have them associated with the current Student Application.
The files will be uploaded once they are added to the Form.IO File component, and the server will create the database relationship between the files and the Student Application when the application is submitted.
- Added server-side validation for
    - Allowed file extensions;
    - Maximum file size;
    - Maximum upload parts;
- Created a **FileService** to replace the Form.IO default one. It allows the proper authentication/authorization of the file being uploaded and downloaded. Upload and download and now handled by _axios_ and authorized in the same way as the other API call on Vue application.
- Created tables to keep the student files relationships.

The information being persisted on application data looks like below:
image

### CODE REVIEW

File: sources/packages/api/src/route-controllers/student/student.controller.ts [hunk 1]
```diff
@@ -134,4 +155,91 @@ export class StudentController extends BaseController {
...
+      fileFilter: defaultFileFilter,
+    }),
+  )
+  async uploadFile(
+    @UserToken() userToken: IUserToken,
+    @UploadedFile() file: Express.Multer.File,
+    @Body("uniqueFileName") uniqueFileName: string,
+    @Body("group") groupName: string,
+  ): Promise<FileCreateDto> {
+    const student = await this.studentService.getStudentByUserName(
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/student/student.controller.ts:181]: do we need to check student exceptions here?

**Review by ann-aot — APPROVED** [review]
LGTM, Greate work @andrewsignori-aot  👍

**Review by abschwenker — APPROVED** [review]

**Review by guru-aot — APPROVED** [review]
That's awesome amount  of work @andrewsignori-aot . I am really excited to see the end to end working of the files upload in our system. 
On the overall implementation, I know there will be upgrades or enhancement to it, like the allowable extensions, file size limit etc, but overall on the implementation LGTM.
Nice work 👍

File: sources/packages/api/src/route-controllers/student/student.controller.ts (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/student/student.controller.ts:181]: Yes, I will add a check.

Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/student/student.controller.ts]: Checks added.

**Comment by andrewsignori-aot** [comment]
> That's awesome amount of work @andrewsignori-aot . I am really excited to see the end to end working of the files upload in our system.
> On the overall implementation, I know there will be upgrades or enhancement to it, like the allowable extensions, file size limit etc, but overall on the implementation LGTM.
> Nice work 👍

Yes, still room for improvements 😃. About the allowed extensions and file size limits, it is implemented on client-side and server side. Did you find any gap on it?

**Pull request merged**